builder/left_recursion_test.go

```diff
@@ -0,0 +1,141 @@
+package builder_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/mna/pigeon/ast"
+	"github.com/mna/pigeon/bootstrap"
+	"github.com/mna/pigeon/builder"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLeftRrecursive(t *testing.T) {
+	t.Parallel()
+	text := `
+	start = expr NEWLINE
```

Comment:
For completeness sake, isn't the definition for `NEWLINE` missing?

Reply:
Fixed